src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerListener.java

```diff
@@ -94,31 +93,29 @@ private void updateDownstreamBuildAction(Run<?, ?> run) {
                 BuildUpstreamCause buildUpstreamCause = (BuildUpstreamCause) cause;
                 Run<?, ?> upstream = buildUpstreamCause.getUpstreamRun();
                 if (upstream instanceof FlowExecutionOwner.Executable) {
-                    FlowExecutionOwner owner = ((FlowExecutionOwner.Executable) upstream).asFlowExecutionOwner();
-                    if (owner == null) {
-                        LOGGER.log(Level.FINE, () -> "Unable to update DownstreamBuildAction for " + upstream + " node " + buildUpstreamCause.getNodeId());
-                        continue;
+                    DownstreamBuildAction action = findMatchingAction(upstream, buildUpstreamCause.getNodeId());
+                    if (action == null) {
+                        action = new DownstreamBuildAction(buildUpstreamCause.getNodeId(), run.getParent());
+                        run.addAction(action);
                     }
+                    action.setBuild(run);
                     try {
-                        FlowExecution execution = owner.get();
-                        FlowNode node = execution.getNode(buildUpstreamCause.getNodeId());
-                        if (node == null) {
-                            LOGGER.log(Level.FINE, () -> "Unable to update DownstreamBuildAction for " + upstream + " node " + buildUpstreamCause.getNodeId());
-                            continue;
-                        }
-                        DownstreamBuildAction downstreamAction = node.getPersistentAction(DownstreamBuildAction.class);
-                        if (downstreamAction == null) {
-                            // Should only happen for builds already in the queue when this plugin is updated to include DownstreamBuildAction.
-                            downstreamAction = new DownstreamBuildAction(run.getParent());
-                            node.addAction(downstreamAction);
-                        }
-                        downstreamAction.setBuild(run);
-                        run.save();
+                        upstream.save();
+                        LOGGER.log(Level.WARNING, () -> "Saved DownstreamBuildAction for " + upstream + " node " + buildUpstreamCause.getNodeId());
```

Comment:
```suggestion
                        LOGGER.fine(() -> "Saved DownstreamBuildAction for " + upstream + " node " + buildUpstreamCause.getNodeId());
```
(use `LoggerRule` from tests as needed)